inc/hooksmigration.class.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ * Formcreator is a plugin which allows creation of custom forms of
+ * easy access.
+ * ---------------------------------------------------------------------
+ * LICENSE
+ *
+ * This file is part of Formcreator.
+ *
+ * Formcreator is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Formcreator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Formcreator. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ * @copyright Copyright © 2011 - 2021 Teclib'
+ * @license   http://www.gnu.org/licenses/gpl.txt GPLv3+
+ * @link      https://github.com/pluginsGLPI/formcreator/
+ * @link      https://pluginsglpi.github.io/formcreator/
+ * @link      http://plugins.glpi-project.org/#/plugin/formcreator
+ * ---------------------------------------------------------------------
+ */
+
+class PluginFormcreatorHooksMigration
+{
+   public static function createSimpleTargetTable($table) {
+      global $DB;
+
+      if ($DB->tableExists($table)) {
+         return;
+      }
+
+      $query = "CREATE TABLE $table (
+         `id` int(11) NOT NULL auto_increment,
+         `name` varchar(255) collate utf8_unicode_ci default NULL,
+         `plugin_formcreator_forms_id` int(11) NOT NULL default '0',
+         `show_rule` int(11) NOT NULL DEFAULT '1',
+         PRIMARY KEY (`id`)
+      ) ENGINE=InnoDB  DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci";
+
+      $DB->queryOrDie($query, "Can't create $table: ". $DB->error());
+   }
```

Comment:
This create table statmeent must be in the plugin_formcreator_empty.sql file with a IF NOT EXIST condition.

This way, upgrade will handle it.

Reply:
No tables were added, this is only an helper method to be used by others plugins.
This allows them to easily create the 'base' target table in one line of code.

They can then add their specific fields manually.